FirebaseDynamicLinks/Sources/FIRDynamicLinks.m

```diff
@@ -419,7 +419,9 @@ - (nullable FIRDynamicLink *)dynamicLinkFromCustomSchemeURL:(NSURL *)url {
                  FDLSDKVersion:FIRFirebaseVersion()
                     completion:^(NSURL *_Nullable resolverURL, NSError *_Nullable resolverError) {
                       if (completion) {
-                        completion(dynamicLink, resolverError);
+                        dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
nit: The completion is called further in the method on potentially a different queue. Maybe we may wrap the completion at the beginning of the method to help to prevent such an issue from being introduced accidentally in the future. E.g.:
```
FIRDynamicLinkUniversalLinkHandler mainQueueHandler = ^{
  if (completion) {
      dispatch_async(dispatch_get_main_queue(), ^{
         completion(dynamicLink, resolverError);
   });
}
```

Reply:
That might be part of a bigger change, since this is now following the pattern of the existing `handleUniversalLink` below.

Reply:
I'm not too strong about it but as for me defining the wrapper and using it instead of the raw completion in the method is a small change that reduces amount of the code and improves readability. I'm fine with leaving it as is as well.

Reply:
let me make a PR. Sorry if my explanation looks confusing.